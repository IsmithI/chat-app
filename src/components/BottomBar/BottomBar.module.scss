@import "../../styles/shadows";
@import "../../styles/colors";
@import "../../styles/shapes";
@import "../../styles/animations";

.container {
	height: spacing(6);
	width: 100vw;
	position: fixed;
	bottom: 0;
	left: 0;

	padding: spacing(1);
	box-sizing: border-box;

	display: flex;
	flex-wrap: nowrap;
	justify-content: space-around;
	align-items: center;

	@include shadow();
	
	border-top-left-radius: $radiusLarge;
	border-top-right-radius: $radiusLarge;
	background-color: #fff;

	z-index: 999;
}

.item {
	padding: spacing(1);
	font-size: 24px;
	color: $darkGrey;

	&.active {
		color: $primaryLight;
		font-size: 32px;
	}

	transition: font-size 0.3s cubic-bezier(.58,.01,.61, 2);
}