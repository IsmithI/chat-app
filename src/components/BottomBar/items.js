import { faBars, faCommentDots, faUserFriends, faCog } from "@fortawesome/free-solid-svg-icons";

export default [
	{
		label: 'menu',
		icon: faBars,
	},
	{
		label: 'chats',
		icon: faCommentDots,
	},
	{
		label: 'contacts',
		icon: faUserFriends,
	},
	{
		label: 'settings',
		icon: faCog,
	},
]