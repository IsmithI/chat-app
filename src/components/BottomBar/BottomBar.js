import React from "react";
import { useState } from "react";
import styles from "./BottomBar.module.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import items from "./items";
import cn from "classnames";

export function BottomBar() {
	const [activeTab, setActiveTab] = useState("chats");

	return (
		<nav className={styles.container}>
			{items.map(item => (
				<Item
					key={item.label}
					{...item}
					onClick={setActiveTab}
					isActive={item.label === activeTab}
				/>
			))}
		</nav>
	);
}

function Item({ label, icon, onClick, isActive }) {
	const classes = cn(styles.item, {
		[styles.active]: isActive
	});

	return (
		<div key={label} onClick={() => onClick(label)} className={classes}>
			<FontAwesomeIcon icon={icon} />
		</div>
	);
}
