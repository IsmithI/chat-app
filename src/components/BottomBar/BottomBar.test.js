import React from 'react';
import renderer from 'react-test-renderer';
import { BottomBar } from './BottomBar';

test('BottomBar renders without errors', () => {
	const component = renderer.create(
		<BottomBar/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
	
})