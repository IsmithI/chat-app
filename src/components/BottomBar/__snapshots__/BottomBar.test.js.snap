// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomBar renders without errors 1`] = `
<nav
  className="container"
>
  <div
    className="item"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bars fa-w-14 "
      data-icon="bars"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="item active"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-comment-dots fa-w-16 "
      data-icon="comment-dots"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32zM128 272c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-user-friends fa-w-20 "
      data-icon="user-friends"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cog fa-w-16 "
      data-icon="cog"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</nav>
`;
